R/stats_utils.R

```diff
@@ -14,18 +14,30 @@ get_stat_summary <- function(obj, name) {
     if (length(obj_class) == 1 && obj_class %in% intrinsic_types) {
         stats[[name]] <- get_stats_column(obj)
     } else if (is.data.frame(obj)) {
-        for(i in 1:ncol(obj)) {
-            col <- obj[,i]
-            col_name <- colnames(obj)[i]
-            stats[[col_name]] <- get_stats_column(col)
+        if (is.null(column)) {
+            for(i in 1:ncol(obj)) {
+                col <- obj[,i]
+                col_name <- colnames(obj)[i]
+                stats[[col_name]] <- get_stats_column(col)
+            }
+        } else if (column %in% colnames(obj)) {
+            col <- obj[,column]
+            stats[[column]] <- get_stats_column(col)
         }
+        
     } else if (length(obj_class) == 1 && is.list(obj)) {
         stats[[name]] <- get_stats_column(obj)
     } else if (is.matrix(obj) && length(dim(obj)) == 2) {
-        for (i in 1:ncol(obj)) {
-            col <- obj[,i]
-            col_name <- as.character(i)
-            stats[[col_name]] <- get_stats_column(col)
+        if (is.null(column)) {
+            for (i in 1:ncol(obj)) {
+                col <- obj[,i]
+                col_name <- as.character(i)
+                stats[[col_name]] <- get_stats_column(col)
+            }
+        } else if (is.numeric(column) && column >= 0 && column < ncol(obj)) {
+            col <- obj[,as.integer(column) + 1]
+
+            stats[[as.character(column + 1)]] <- get_stats_column(col)
```

Comment:
I feel like this should respect R's 1-based indexing, the client is gonna get column names "1, 2, 3" for var details, passing in col "1" for a stats summary should return the stats for the first column

Reply:
I think you're right- I was going back and forth on that as I was testing it